docs/DeveloperGuide.md

```diff
@@ -1,34 +1,247 @@
+## 2. Design
+![image_info](./pictures/ClassDiagram.png)
+_Fig 2.1. Class diagram of the Jikan program_
 
 
-## Product Scope
-### Target user profile
+## 3. Implementation
 
-{Describe the target user profile}
+This section describes some noteworthy details on how certain features are implemented.
 
-### Value proposition
+### 3.1 Automated Storage Cleanup feature
 
-{Describe the value proposition: what problem does it solve?}
+#### 3.1.1 Current Implementation
 
-## User Stories
+The storage cleanup mechanism is stored internally as a StorageCleaner java class. The StorageCleaner class has an association with the Storage class and thus it is able to access and edit the datafile which contains the list of activities. 
+
+Additionally, when the StorageCleaner class is first initialised, it will create two files namely a status file and a data file in the “recycled” folder under the “data” folder. The status file keeps track of the activation status of the storage cleaner while the data file serves as a recycle bin for deleted data. 
+
+Moreover, the class also implements the following operations:
+
+`StorageCleaner#initialiseCleaner - Loads up the status file to decide whether to activate/deactivate the automated cleaner.`
+
+`StorageCleaner#setStatus - Gives the user the freedom to activate/de-activate the cleaner using commands during application runtime.`
+
+`StorageCleaner#autoClean - This operation is called whenever the application is executed. Storage cleaning will only be done if the cleaner is activated.`
+
+Given below is the example scenario of how the operations work.
+
+##### initialiseCleaner
+
+Step 1. Loads up the status file for reading. If the status file is not found, create a new status file and write the character ‘0’ to the first line of the status file.
+
+Step 2. Read the first line of the status file, if a character ‘0’ is found, deactivate the automated cleanup. Else if a character ‘1’ is found, activate the automated cleanup.
+
+![image_info](./pictures/FlowchartinitCleaner.png)
+
+##### setStatus
+
+Step 1. Read the boolean parameter ‘status’.
+
+Step 2a. If ‘status’ is equal to true, activate the automated cleanup and write the character ‘1’ to the first line of the status file. (overwrite any existing value at the first line).
+
+Step 2b. If ‘status’ is equal to false, deactivate the automated cleanup and write the character ‘0’ to the first line of the status file.
+
+![image_info](./pictures/FlowchartsetStatus.png)
+
+##### autoClean
+
+Step 1. Check activation status of StorageCleaner through the class level attribute boolean toClean.
+
+Step 2a. If the attribute toClean is equal to false, return the operation and give control back to the caller.
+
+Step 2b. If the attribute toClean is equal to true, access the storage data file and remove some of the activities starting from the oldest. Put these deleted activities into the data file under the ‘recycled’ folder.
+
+![image_info](./pictures/SDautoClean.png)
```

Comment:
The dotted lines from the Storage Class and StorageCleaner class extends beyond the cross symbol. Perhaps you could change the dotted lines to end at the cross symbol since the classes would already have been deleted by then?